scipy/stats/tests/test_stats.py

```diff
@@ -4511,10 +4544,26 @@ def test_normalitytests():
 
 
 class TestRankSums(object):
-    def test_ranksums_result_attributes(self):
-        res = stats.ranksums(np.arange(5), np.arange(25))
-        attributes = ('statistic', 'pvalue')
-        check_named_results(res, attributes)
+
+    np.random.seed(0)
+    x, y = np.random.rand(2, 10)
+
+    @pytest.mark.parametrize('alternative', ['less', 'greater', 'two-sided'])
+    def test_ranksums_result_attributes(self, alternative):
+        # ranksums pval = mannwhitneyu pval w/out continuity or tie correction
```

Comment:
@tirthasheshpatel It seems there was no need to use that data, since we didn't have an external p-value to compare against. In that case, why not just use something without ties so that the p-values would be the same?

Do you agree that this does the same job?

Also, splitting things into separate functions, each with a particular job, makes it a little easier to read IMO.

Reply:
> Do you agree that this does the same job?

Yes, this should be equivalent to what we had before + this is simpler.

> Also, splitting things into separate functions, each with a particular job, makes it a little easier to read IMO. Do you agree?

This makes sense. It's more readable to me too. Thanks!